pkg/queryfrontend/config.go

```diff
@@ -186,6 +187,7 @@ type DownstreamTripperConfig struct {
 	MaxIdleConns          *int               `yaml:"max_idle_conns"`
 	MaxIdleConnsPerHost   *int               `yaml:"max_idle_conns_per_host"`
 	MaxConnsPerHost       *int               `yaml:"max_conns_per_host"`
+	TLSConfig             *tls.Config        `yaml:"tls_config"`
```

Comment:
I don't think this is the correct struct for TLS configs. Should be `exthttp.TLSConfig`, which you can then process and pass to `downstreamTripper.TLSClientConfig` by using `exthttp.NewTLSConfig()`